legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabEntityApi.java

```diff
@@ -714,11 +734,31 @@ private Stream<EntityProjectFile> getEntityProjectFiles(ProjectFileAccessProvide
         }
         if (classifierPathPredicate != null)
         {
-            stream = stream.filter(epf -> classifierPathPredicate.test(epf.getEntity().getClassifierPath()));
+            stream = stream.filter(excludeInvalid ? epf ->
+            {
+                try
+                {
+                    return classifierPathPredicate.test(epf.getEntity().getClassifierPath());
+                }
+                catch (Exception ignore)
+                {
+                    return false;
+                }
+            } : epf -> classifierPathPredicate.test(epf.getEntity().getClassifierPath()));
         }
         if (contentPredicate != null)
         {
-            stream = stream.filter(epf -> contentPredicate.test(epf.getEntity().getContent()));
+            stream = stream.filter(excludeInvalid ? epf ->
+            {
+                try
+                {
+                    return contentPredicate.test(epf.getEntity().getContent());
+                }
+                catch (Exception ignore)
+                {
+                    return false;
+                }
```

Comment:
See the above comment. Also, we may need to think about caching any exception that may be thrown inside epf.getEntity() so that if it is called a second time the same exception will be thrown. Otherwise, if it's called a second time, we'll probably get some exception about the stream being closed or something.